pkg/controller/node/node_controller.go

```diff
@@ -613,15 +613,20 @@ func (ctrl *Controller) getPoolsForNode(node *corev1.Node) ([]*mcfgv1.MachineCon
 	if len(custom) > 1 {
 		return nil, fmt.Errorf("node %s belongs to %d custom roles, cannot proceed with this Node", node.Name, len(custom))
 	} else if len(custom) == 1 {
-		// We don't support making custom pools for masters
+		pls := []*mcfgv1.MachineConfigPool{}
 		if master != nil {
-			return nil, fmt.Errorf("node %s has both master role and custom role %s", node.Name, custom[0].Name)
+			// if we have a custom pool and master, defer to master and return.
+			glog.Infof("Given both master and custom pools. Defaulting to master: custom %v", custom[0].Name)
+			ctrlcommon.MCCPoolAlert.WithLabelValues(custom[0].Name, fmt.Sprintf("Given both master and custom pools. Defaulting to master: custom %v", custom[0].Name)).SetToCurrentTime()
```

Comment:
Two quick questions for this alert:
1. if we are hitting this, will it keep adding new alerts or set a new time for the same one? Since this function is called whenever a node sync happens, what would the user see in terms of alerts?
2. if the user "fixes" their issue, does the alert stay? Or is it just a one-time thing?

Reply:
Hello!
This is the same behavior that we saw in our verification. No alert was fired. Since expected the alert to be fired, we have not added the qe-approved label.